services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/GenericS3TransferManager.java

```diff
@@ -182,8 +194,109 @@ public FileUpload uploadFile(UploadFileRequest uploadFileRequest) {
         } catch (Throwable throwable) {
             returnFuture.completeExceptionally(throwable);
         }
+        return new DefaultFileUpload(returnFuture, progressUpdater.progress(), pauseObservable, uploadFileRequest);
+    }
+
+    @Override
+    public FileUpload resumeUploadFile(ResumableFileUpload resumableFileUpload) {
+        Validate.paramNotNull(resumableFileUpload, "resumableFileUpload");
+
+        boolean fileModified = !FileUtils.fileNotModified(resumableFileUpload.fileLength(),
```

Comment:
IIRC, `CrtS3TransferManager#resumeUploadFile` has similar logic, can we reuse the code?